### DISCUSSION TITLE
JaversBuilder in CustomPropertyComparator

### DISCUSSION THREAD

**Discussion by ArkadiuszBlejwas** [javers/javers#1073] [Q&A]
I want use JaversBuilder in my Custom Property Comparator as in the example below:

```
public class MyObjectComparator implements CustomPropertyComparator<MyObject, SetChange> {
    
    @Override
    public Optional<SetChange> compare(MyObject myObject, MyObject t1, PropertyChangeMetadata propertyChangeMetadata, Property property) {
        if (equals(myObject, t1)) {
            return Optional.empty();
        } else {
            Javers javers = JaversBuilder.javers().build();
            Diff diff = javers.compare(myObject, t1);
            return Optional.of(diff.getChanges());
        }
    }
//equals
//toString
```
But the return type doesn't match. Changes =/= SetChange. Is there way to use JaversBuilders in Custom Property Comparator?

**Comment by bartoszwalacik** [comment]
I think you don't understand how CustomPropertyComparator works. You are not supposed to embed here another javers instance and delegate diff logic to it. You are supposed to implement the diff logic on your own. 
See https://javers.org/documentation/diff-examples/#custom-property-comparator-example

**Reply by ArkadiuszBlejwas** [reply]
ok. it's a pity that there is no example for a more complex object :/ String is a fairly simple object. Is there any easy way to collect changes after checking equals(I mean snippet after 'else') for more complex objects?

**Reply by bartoszwalacik** [reply]
There is no such example because CustomPropertyComparator is designed to compare Values. Why do you want to compare complex objects with a CustomPropertyComparator?

**Reply by ArkadiuszBlejwas** [reply]
It seemed to me that is good way to change comparing some fields in very complex object. I send to you message on facebook(messanger). Can we talk? You are a contributor of javers and a very smart guy. I think you would be able to help me and I would repay it.

**Reply by bartoszwalacik** [reply]
here is the place to talk about javers
 
javers has 100 000 users, I don't want them all on my priv messanger

**Comment by ranjan5ua** [comment]
Javers is comparing java.sql.Date object wrong. Even if after writing a custom comparator and registering it (somehow), it is still not working. 
What to do?

**Reply by bartoszwalacik** [reply]
@ranjan5ua please stop cross-posting and follow the guidelines you got here https://github.com/javers/javers/issues/1407
